generate/src/contract/methods.rs

```diff
@@ -95,24 +105,12 @@ fn expand_function(cx: &Context, function: &Function) -> Result<TokenStream> {
     Ok(quote! {
         #doc
         pub fn #name(&self #input) -> #result {
-            self.instance.#method(#name_str, #arg)
+            self.instance.#method(#signature_lit, #arg)
                 .expect("generated call")
         }
     })
 }
 
-fn function_signature(function: &Function) -> String {
-    let types = match function.inputs.len() {
-        0 => String::new(),
-        _ => {
-            let mut params = function.inputs.iter().map(|param| &param.kind);
-            let first = params.next().expect("at least one param").to_string();
-            params.fold(first, |acc, param| format!("{},{}", acc, param))
-        }
-    };
-    format!("{}({})", function.name, types)
-}
-
```

Comment:
Any particular reason this is no longer needed?

Reply:
This was implemented twice in two separate places, here and in `ethcontract_common`. I just removed this implementation and used the shared one instead.